src/node/interfaces.cpp

```diff
@@ -653,24 +654,28 @@ class ChainImpl : public Chain
     }
     void getPackageLimits(unsigned int& limit_ancestor_count, unsigned int& limit_descendant_count) override
     {
-        limit_ancestor_count = gArgs.GetIntArg("-limitancestorcount", DEFAULT_ANCESTOR_LIMIT);
-        limit_descendant_count = gArgs.GetIntArg("-limitdescendantcount", DEFAULT_DESCENDANT_LIMIT);
+        if (!m_node.mempool) {
+            const CTxMemPool::Limits limits{};
+            limit_ancestor_count = limits.ancestor_count;
+            limit_descendant_count = limits.descendant_count;
+        } else {
+            const CTxMemPool::Limits& limits{m_node.mempool->m_limits};
+            limit_ancestor_count = limits.ancestor_count;
+            limit_descendant_count = limits.descendant_count;
+        }
```

Comment:
nit in b810f102c47c61f276ffa2f9a78a3b86ad5a88b8 for less code:

```suggestion
            const CTxMemPool::Limits default_limits{};
            
            const CTxMemPool::Limits& limits{m_node.mempool?m_node.mempool->m_limits:default_limits};
            limit_ancestor_count = limits.ancestor_count;
            limit_descendant_count = limits.descendant_count;
```

Reply:
Fixed as of d669b93469